### ISSUE TITLE
resource `okta_user_schema_property` having any `array_type` other than "string" and `array_enum` values

### ISSUE THREAD

**Issue by monde** [okta/terraform-provider-okta#1074]
There is a known issue with the Okta API underpinning the resource `okta_user_schema_property`. If the resource has a `array_type` other than "string" and an `array_enum` of values, then the Okta API will incur 500 errors. For example the [array_number.tf](https://github.com/okta/terraform-provider-okta/blob/master/examples/okta_user_schema_property/array_number.tf):
```terraform
resource "okta_user_schema_property" "test" {
  index       = "testAcc_replace_with_uuid"
  title       = "terraform acceptance test"
  type        = "array"
  description = "testing"
  master      = "OKTA"
  scope       = "SELF"
  array_type  = "number"
  array_enum  = [0.01, 0.02, 0.03]
  array_one_of {
    title = "1"
    const = 0.01
  }
  array_one_of {
    title = "2"
    const = 0.02
  }
  array_one_of {
    title = "3"
    const = 0.03
  }
}
```
Will cause this kind of error:
```
[ERROR] the API returned an error: The enum constraint com.fasterxml.jackson.databind.node.TextNode "0.01" is not valid for the given property type java.lang.Number., x-okta-request-id=XXX
```

**Comment by monde** [comment]
This goes back to the SDK
https://github.com/okta/okta-sdk-golang/issues/304

**Comment by monde** [comment]
Fixed in [v3.30.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.30.0)
